src/asm/parser.y

```diff
@@ -319,13 +321,13 @@ static void strfmt(char *dest, size_t destLen, char const *fmt, std::vector<stru
 			continue;
 		}
 
-		struct StrFmtArg &arg = args[a++];
+		std::variant<uint32_t, char *> &arg = args[a++];
 		static char buf[MAXSTRLEN + 1];
 
-		if (arg.isNumeric)
-			fmt_PrintNumber(buf, sizeof(buf), &spec, arg.number);
+		if (std::holds_alternative<uint32_t>(arg))
+			fmt_PrintNumber(buf, sizeof(buf), &spec, std::get<uint32_t>(arg));
 		else
-			fmt_PrintString(buf, sizeof(buf), &spec, arg.string);
+			fmt_PrintString(buf, sizeof(buf), &spec, std::get<char *>(arg));
```

Comment:
This is not bad code by any metric, but I wonder if `visit` wouldn't be more appropriate here?

Reply:
Arguably yes. In this case it's just two alternatives, and I'd have had to move the `Visitor` template helper into a utility file both rgbgfx and rgbasm could use.

Reply:
Switched to using `visit`.